pkg/minikube/machine/fix.go

```diff
@@ -145,6 +150,26 @@ func fixHost(api libmachine.API, mc config.ClusterConfig) (*host.Host, error) {
 	return h, ensureSyncedGuestClock(h, mc.Driver)
 }
 
+// maybeWarnAboutEvalEnv wil warn user if they need to re-eval their docker-env, podman-env
+// because docker changes the allocated bind ports after restart https://github.com/kubernetes/minikube/issues/6824
+func maybeWarnAboutEvalEnv(drver string, name string) {
+	if !driver.IsKIC(drver) {
+		return
+	}
+	p := os.Getenv(constants.MinikubeActiveDockerdEnv)
+	if p == "" {
+		return
+	}
+	out.T(out.Notice, "Noticed that you are using minikube docker-env:")
+	out.T(out.WarningType, `After minikube restart the dockerd ports might have changed. To ensure docker-env works properly.
+Please re-eval the docker-env command:
```

Comment:
nit: "Please rerun the docker-env command:"

Reply:
I was thinking re run wouldn't actually help. They have to actually do the eval 
But since it is different for each bash I left that part to them.

Ideally we want to give them the right eval command based on their shell